PF/Assignment1/TrainMenu.java

```diff
@@ -0,0 +1,161 @@
+import java.util.*;
+
+public class TrainMenu {
+
+	public void menu() {
+		Scanner sc = new Scanner(System.in);
+		int tempVar;
+		do {
+			int count = 0;
+
+			String fromPlace, toPlace;
+			System.out.println("Enter your name");
+
+			String userName = sc.next();
+			MenuList menulist = new MenuList();
+			int trainType;
+			do {
+				menulist.displayMenu();
+				trainType = sc.nextInt();
+				if ((trainType != 1) && (trainType != 2)) {
+					System.out.println("Enter right values");
+
+				}
+			} while ((trainType != 1) && (trainType != 2));
+			Procedure procedure = new Procedure();
+			procedure.displayChart(trainType);
+
+			int check = 0;
+			do {
+
+				System.out.println("Enter from where are you going");
+				fromPlace = sc.next();
+				System.out.println("Enter where are you going");
+				toPlace = sc.next();
+				check = procedure
+						.availableTrains(trainType, fromPlace, toPlace);
+			} while (check != 1);
+
+			if (trainType == 1) {
+				System.out.println("Enter train id ");
+				int trainid = sc.nextInt();
+
+				Iterator<PassengerTrain> itr = TrainDetails.passengerList
+						.iterator();
+				while (itr.hasNext()) {
+					PassengerTrain object1 = itr.next();
+					if ((object1.trainId == trainid)) {
+						System.out.println("Enter no of seats");// tracking
+																// seats
+						int seats = sc.nextInt();
+						if (seats < object1.seats) {
+							double amount = seats * object1.price;
+							int newSeats = object1.seats - seats;
```

Comment:
ACTION: You can directly set new seats instead of creating new variable. Remove this variable and assign new seat directly.